sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -327,12 +347,12 @@ private static void handleAppStartProfilingConfig(
 
   @SuppressWarnings("FutureReturnValueIgnored")
   private static void finalizePreviousSession(
-      final @NotNull SentryOptions options, final @NotNull IHub hub) {
+      final @NotNull SentryOptions options, final @NotNull IScopes scopes) {
     // enqueue a task to finalize previous session. Since the executor
     // is single-threaded, this task will be enqueued sequentially after all integrations that have
     // to modify the previous session have done their work, even if they do that async.
     try {
-      options.getExecutorService().submit(new PreviousSessionFinalizer(options, hub));
+      options.getExecutorService().submit(new PreviousSessionFinalizer(options, scopes));
```

Comment:
Could use the `HubScopesWrapper` here to make it compile:
`options.getExecutorService().submit(new PreviousSessionFinalizer(options, new HubScopesWrapper(scopes)));`

Reply:
done in a follow up pr